packages/firestore/src/api/credentials.ts

```diff
@@ -179,7 +182,13 @@ export class FirebaseCredentialsProvider implements CredentialsProvider {
     const initialUserCounter = this.userCounter;
     const forceRefresh = this.forceRefresh;
     this.forceRefresh = false;
-    return (this.app as _FirebaseApp).INTERNAL.getToken(forceRefresh).then(
+    var token: Promise<FirebaseAuthTokenData>;
```

Comment:
Do we have to modify the Firestore SDK in order to get this to work? I'd prefer not to introduce this `tokenOverride` value.

Reply:
AFAICT, we'd have to change `app-types` to export `FirebaseAppImpl` or to move `INTERNAL` into `FirebaseApp` to give visibility to the testing package. If that's the only way to do this, I'd rather modify the `firestore` package than `app` package